### PR TITLE
[Project] Take project source into account when running `build_image`

### PR DESCRIPTION
**Pull request by theSaarco**
The `project.build_image()` function did not take into account the project source. If the user is setting a source to the project (for example `git` source) and set `pull_at_runtime = False`, then it is expected that images used would contain the project source in them. This worked for specific functions when building them, but not in the project level.
The expectation is that running this code:
```
project = mlrun.get_or_create_project(
    name="project-name", context="./"
)

project.set_source(
    "git://some/repo",
    pull_at_runtime=False
)

project.build_image(image=".some-project-image")
```
Would build the `.some-project-image` image with the source in it, such that running a function will work without having to specify an image:
```
func = project.set_function(handler="package.function", name="func", kind="job")
func.save()
project.run_function("func", params={...})
```
This would work even without having the source code available on the dev site, and will not require pulling the code at execution time.

This fixes [ML-5481](https://jira.iguazeng.com/browse/ML-5481)

### CODE REVIEW

File: mlrun/projects/project.py [hunk 1]
```diff
@@ -3115,6 +3115,13 @@ def build_image(
 
             function = mlrun.new_function("mlrun--project--image--builder", kind="job")
 
+            if self.spec.source and not self.spec.load_source_on_run:
+                function.with_source_archive(
+                    source=self.spec.source,
+                    workdir=self.spec.workdir,
```
Comment (alonmr) [inline mlrun/projects/project.py:3121]: workdir is not part of the build job but it is set during runtime (before starting the run)

File: tests/projects/test_project.py [hunk 2]
```diff
@@ -47,6 +48,37 @@ def assets_path():
     return pathlib.Path(__file__).absolute().parent / "assets"
 
 
+@pytest.fixture
+def remote_builder_mock(monkeypatch):
+    class _MockDb:
```
Comment (alonmr) [inline tests/projects/test_project.py:53]: why setting the class inside the fixture? it can be done like our k8s secrets mock

**Review by alonmr — COMMENTED** [review]
Nice! Minor comments

File: mlrun/projects/project.py (hunk 1 above)
Comment (alonmr) [reply mlrun/projects/project.py:3121]: huh you are right :( that's my fault.. we should move it to the build spec

File: tests/projects/test_project.py (hunk 2 above)
Comment (theSaarco) [reply tests/projects/test_project.py]: Fixed.

File: mlrun/projects/project.py (hunk 1 above)
Comment (theSaarco) [reply mlrun/projects/project.py]: It's also not part of `project.set_source`, and it does make sense. I think for the project use-case, it's ok to have it as a parameter for `build_image`, which is what I did.

**Review by alonmr — APPROVED** [review]

**Pull request merged**